client/integration-tests/equals-hashcode/src/test/java/io/quarkiverse/openapi/generator/it/VerifyGenerationTest.java

```diff
@@ -0,0 +1,47 @@
+package io.quarkiverse.openapi.generator.it;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.stream.Stream;
+
+import org.acme.non.equals.hashcode.model.Animal;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import io.quarkus.test.junit.QuarkusTest;
+
+@QuarkusTest
+class VerifyGenerationTest {
+
+    @ParameterizedTest
+    @MethodSource("provideStringsForIsBlank")
+    void verify(Class<?> clazz, String methodName, Class<?> expectedDeclaringClass) {
+        var equalsMethod = getMethod(clazz, methodName);
+
+        Class<?> declaringClass = equalsMethod.getDeclaringClass();
+
+        assertEquals(declaringClass, expectedDeclaringClass);
+    }
+
+    private static Stream<Arguments> provideStringsForIsBlank() {
+        return Stream.of(
+                Arguments.of(Animal.class, "equals", Object.class),
+                Arguments.of(Animal.class, "hashCode", Object.class),
```

Comment:
Do we really need to check `Object`?

Reply:
I thought a long time of how to verify that we don't generate `equals`and ´hashCode` and the best way I thought out was to get what class that declares a method. This way I can check if the generated model has or does not have the methods based on what set in the `application.properties`.

In the case of the models that doesn't have a generated `equals` and `hashCode`, the methods are declared in ´Object´.

What I can do is I can split the tests and don't check against `Object` and instead just check that the methods its not declared in the generated model. Or I can leave it like this?

Or does somebody have an better way of verifying the generation?

Reply:
My point is that `Object` will always have `equals` and `hashCode` declared. The verification will always pass.

Reply:
Thats a valid point @hbelmiro, I now see the problem with my test. I have rewritten the test so it only tests the models instead.